File: composer/src/DTOs/UIPageConfig.php
<?php

namespace ReactUbiquitous\DTOs;

use ReactUbiquitous\Contracts\SerializableInterface;

final class UIPageConfig implements SerializableInterface
{
    /** @param array $sections UISectionConfig[] */
    public function __construct(
        public readonly string $id,
        public readonly string $title,
        public readonly int $order,
        public readonly array $sections,
        public readonly ?string $description = null,
        public readonly ?string $icon = null,
        public readonly ?string $className = null,
        public readonly ?bool $visible = null,
    ) {}

    public function toArray(): array
    {
        return array_filter([
            'id' => $this->id,
            'title' => $this->title,
            'description' => $this->description,
            'icon' => $this->icon,
            'order' => $this->order,
            'sections' => array_map(fn($s) => $s->toArray(), $this->sections),
            'className' => $this->className,
            'visible' => $this->visible,
        ], fn($v) => $v !== null);
    }
}

File: nuget/src/ReactUbiquitous.NuGet/Models/UIPageConfig.cs
using System.Text.Json.Serialization;
using ReactUbiquitous.NuGet.Models.Sections;

namespace ReactUbiquitous.NuGet.Models;

/// <summary>Represents the configuration for a single UI page.</summary>
public class UIPageConfig
{
    /// <summary>Gets or sets the unique page identifier.</summary>
    [JsonPropertyName("id")]
    public string Id { get; set; } = string.Empty;

    /// <summary>Gets or sets the page title.</summary>
    [JsonPropertyName("title")]
    public string Title { get; set; } = string.Empty;

    /// <summary>Gets or sets the page description.</summary>
    [JsonPropertyName("description")]
    public string? Description { get; set; }

    /// <summary>Gets or sets the icon associated with the page.</summary>
    [JsonPropertyName("icon")]
    public string? Icon { get; set; }

    /// <summary>Gets or sets the display order of the page.</summary>
    [JsonPropertyName("order")]
    public int Order { get; set; }

    /// <summary>Gets or sets the list of sections contained in the page.</summary>
    [JsonPropertyName("sections")]
    public List<BaseSectionConfig>? Sections { get; set; }

    /// <summary>Gets or sets additional CSS class names for the page.</summary>
    [JsonPropertyName("className")]
    public string? ClassName { get; set; }

    /// <summary>Gets or sets a value indicating whether the page is visible.</summary>
    [JsonPropertyName("visible")]
    public bool? Visible { get; set; }
}
